### ISSUE TITLE
Selenium tests fail to run due to TestNG errors

### ISSUE THREAD

**Issue by hnestorova** [cbeust/testng#2066]
### TestNG Version
6.14.3

### Expected behavior
After Selenium upgrade to version 3.141.59, the tests are successfully run from Jenkins v2.138.3 as they used to run with lower Selenium version

### Actual behavior
After Selenium upgrade to version 3.141.59, the tests fail to run with TestNG on Jenkins v2.138.3 errors like:
TestNGClassFinder: Warning: Can't link and determine methods of class....
TestNGClassFinder: Unable to read methods on class ... -unable to resolve class reference org/apache/http/HttpRequest

### Is the issue reproductible on runner?

- [ ] Shell
- [ ] Maven
- [ ] Gradle
- [ ] Ant
- [X] Eclipse : when trying to run the tests from Eclipse 2019-03 , I get a message that either TestNG cannot find class in classpath or that No tests were found and nothing was run
- [ ] IntelliJ
- [ ] NetBeans

### Test case sample
See attached screenshots. Note that I have marked certain parts as Private due to security. However, I am 100% certain the classpath is correct.
![Java_class](https://user-images.githubusercontent.com/30794708/56846674-a3b1ab80-68da-11e9-9a42-828c8a1e5f2f.PNG)
![TestNG_error](https://user-images.githubusercontent.com/30794708/56846675-a44a4200-68da-11e9-89ff-9946ef4b6e30.PNG)

**Comment by krmahadevan** [comment]
The error that you have shared `cannot find class in classpath` is usually TestNG's way of telling you that the fully qualified name that you provided (packagename.classname) is perhaps incorrect and doesn't represent a valid class in the class path. 

There's not much that TestNG can do here. You need to fix the issue by ensuring that you provide the correct classname.

Are you getting this error when you right click a class and choose `Run As > TestNG test` ? If yes, then I would suggest that you please create a bug report in the [TestNG eclipse project](https://github.com/cbeust/testng-eclipse) instead of creating it here. 

For the second issue viz., 

> After Selenium upgrade to version 3.141.59, the tests are successfully run from Jenkins v2.138.3 as they used to run with lower Selenium version

TestNG doesn't have any control on this one. You would need to fix your CLASSPATH issues.

**Comment by hnestorova** [comment]
@krmahadevan  this one whould not be closed. Hi,

Its not a classpath issue - the tests with the same classpaths run suceessfully with a lower selenium version. Pls check the bug report for what i get when i run the tests from eclipse again only with tje newer selenium version

**Comment by krmahadevan** [comment]
@hnestorova - Unless and until you can provide a simple standalone test project (Which does not involve selenium) that can be used to recreate the problem, there's not much that can be done here.

**Comment by hnestorova** [comment]
Eclipse : when trying to run the tests from Eclipse 2019-03 , I get a message that either TestNG cannot find class in classpath or that No tests were found and nothing was run
with thelower selenium version i dont get the problem

**Comment by hnestorova** [comment]
what would this project look  like and why without selenium when i have the probablem exactly whwn i use it for such tests?

**Comment by krmahadevan** [comment]
@hnestorova - Please take a look at my comment https://github.com/cbeust/testng/issues/2066#issuecomment-488219622

You are only telling us what flavor of eclipse you are working with. You haven't provided the following

1. What version of TestNG eclipse plugin are you working with ?
2. How are you running the tests ? 
3. What does the java project structure look like ?

**Comment by krmahadevan** [comment]
@hnestorova 

> what would this project look like and why without selenium when i have the probablem exactly whwn i use it for such tests?

Because TestNG doesn't have anything to do with Selenium and we cannot resolve class path related issues which are arising due to a selenium version upgrade in dependency.

**Comment by hnestorova** [comment]
1. I am using the latest testng plugin version
2. either from eclipse - run as testng or from Jenkins
3. i have provided a java class with a sample teta case and a corresponding testng xml

**Comment by hnestorova** [comment]
@krmahadevan there are quire a few users that do use selenium and testng together so i do þink it is good to keepmthem compatible

**Comment by krmahadevan** [comment]
@hnestorova 

> I am using the latest testng plugin version

Please mention the version number.

> i have provided a java class with a sample teta case and a corresponding testng xml

No you haven't. You have shared a screenshot of the code. Not sure what do you expect us to do with a screenshot? I reiterate what I asked earlier. Please provide a sample project (preferably a maven/gradle project) which contains a sample test class that can be used to recreate the issue. 

> there are quire a few users that do use selenium and testng together so i do þink it is good to keepmthem compatible

Sure. But why do you expect TestNG to provide the compatibility ? TestNG merely requires `JCommander` as a hard required dependency. The rest is all being brought by Selenium.

**Comment by hnestorova** [comment]
OK, I thought the idea of forum discussions was to resolve problems and
help others. Instead, it seems people should feel guilty they posted their
problem :) if you decide you can assist me in resolving the issue, let me
know and lets start working on it instead of pointing fingers. Have a nice
day

На чт, 2.05.2019 г. в 8:52 ч. Krishnan Mahadevan <notifications@github.com>
написа:

**Comment by krmahadevan** [comment]
> OK, I thought the idea of forum discussions was to resolve problems and help others. Instead, it seems people should feel guilty they posted their problem :) 

Lets not get melodramatic here. My intention is to help you fix the problem. But you need to provide what I am asking for. And the ask is merely a sample standalone project which can be used to reproduce the problem.  This is the basics of any bug. Provide a sample to recreate the issue.
Not sure why you seem to be hesitant in providing that?  I believe the same was echoed to [your post](https://groups.google.com/forum/#!topic/selenium-users/oZAepU7SpYk) on the selenium google forums as well 

>  if you decide you can assist me in resolving the issue, let me know and lets start working on it instead of pointing fingers. Have a nice day

Let me state the same thing. If you decide you would like to get your issue resolved by providing a sample project to start off with, let me know and I will gladly walk you through on what the problem is. 

But its not going to be possible for me or anyone else in the TestNG dev group for that matter to guess everything and resolve the problem for you.

To summarise I need the following :

1. TestNG eclipse version being used.
2. A simple standalone test project that can be used to reproduce the issue [ The project should contain the testing suite xml and also the test classes that are needed ]
3. Any additional information which you feel can be relevant to your problem and can help expedite the resolution.

Good day to you as well !!
